packages/aws-cdk-lib/aws-route53/lib/health-check.ts

```diff
@@ -0,0 +1,517 @@
+   *
+   * @default 3
+   */
+  readonly failureThreshold?: number;
+
+  /**
+   * Fully qualified domain name of the endpoint to be checked.
+   *
+   * Amazon Route 53 behavior depends on whether you specify a value for IPAddress.
+   *
+   * If you specify a value for IPAddress:
+   *
+   * Amazon Route 53 sends health check requests to the specified IPv4 or IPv6 address and passes the value of FullyQualifiedDomainName in the Host header for all health checks except TCP health checks. This is typically the fully qualified DNS name of the endpoint on which you want Route 53 to perform health checks.
+   * Note: If you specify a value for Port property other than 80 or 443, Route 53 will constract the value for Host header as FullyQualifiedDomainName:Port.
+   *
+   * If you don't specify a value for IPAddress:
+   *
+   * Route 53 sends a DNS request to the domain that you specify for FullyQualifiedDomainName at the interval that you specify for RequestInterval. Using an IPv4 address that DNS returns, Route 53 then checks the health of the endpoint.
+   *
+   * Additionally, if the type of the health check is HTTP, HTTPS, HTTP_STR_MATCH, or HTTPS_STR_MATCH, Route 53 passes the value of FullyQualifiedDomainName in the Host header, as it does when you specify value for IPAddress. If the type is TCP, Route 53 doesn't pass a Host header.
+   *
+   * @default - not configured
+   */
+  readonly fqdn?: string;
+
+  /**
+   * The number of child health checks that are associated with a CALCULATED health that Amazon Route 53 must consider healthy for the CALCULATED health check to be considered healthy.
+   *
+   * @default - not configured
+   */
+  readonly healthThreshold?: number;
+
+  /**
+   * The status of the health check when CloudWatch has insufficient data about the state of associated alarm.
+   *
+   * @default - not configured
+   */
+  readonly insufficientDataHealthStatus?: InsufficientDataHealthStatusEnum;
+
+  /**
+   * Specify whether you want Amazon Route 53 to invert the status of a health check, so a health check that would normally be considered unhealthy is considered healthy, and vice versa.
+   *
+   * @default - not configured
+   */
+  readonly inverted?: boolean;
+
+  /**
+   * The IPv4 or IPv6 IP address for the endpoint that you want Amazon Route 53 to perform health checks on. If you don't specify a value for IPAddress, Route 53 sends a DNS request to resolve the domain name that you specify in FullyQualifiedDomainName at the interval that you specify in RequestInterval. Using an IPv4 address that DNS returns, Route 53 then checks the health of the endpoint.
+   *
+   * @default - not configured
+   */
+  readonly ipAddress?: string;
+
+  /**
+   * Specify whether you want Amazon Route 53 to measure the latency between health checkers in multiple AWS regions and your endpoint, and to display CloudWatch latency graphs on the Health Checks page in the Route 53 console.
+   *
+   * @default - not configured
```

Comment:
Again, not explicitly said, but I'm going to assume the default is `false`

Reply:
See comment about setting the default value: https://github.com/aws/aws-cdk/pull/30739#discussion_r1671755028